src/move_data_from_insdc.py

```diff
@@ -182,11 +183,27 @@ def define_source_parameters(path: str) -> (any([OpenerDirector, str]), int, str
                      Source of the file. Currently accepted: ftp, s3, local.
     """
     if "ftp" in path:
+        retries = 0
+        source = "ftp"
         if not path.startswith("ftp://"):
             path = f"ftp://{path}"
-        streamable = urlopen(path)
-        file_size = int(streamable.headers['Content-length'])
-        source = "ftp"
+        while True:
```

Comment:
```
for retries in range(10):
    try:
    .
    .
    .
    Except URLError as e:
       print(e.reason)
       print("Retrying...")

        if e.errno == 421:
             print("Waiting for 5 seconds...")
             sleep(5)
else:
    raise IOError(f"Retried the maximum amount of times to get stream from {path}.")

Reply:
I think this is a little bit more clear.

Also, btw, have you tested what happens if you raise that error when there are multiple threads working? does it end only that thread, or does it end the whole pool?

Reply:
Haha no way! I did exactly the same change 30 minutes ago (I just pushed my commit). 

Yes I did, it closes the whole pool. I'm trying to double check now but nothing is erroring!

Reply:
Ah and using the error number does not work. The error number is not part of the URLError unfortunately so string matching will have to suffice

Reply:
ouch! that sucks, I'd expect to get the error number lol

PS: Great minds think alike!